notifier/microsoftteamsnotifier/notifier.go

```diff
@@ -0,0 +1,212 @@
+	attachments := make([]attachment, 0)
+	for key := range diffCache.Deleted {
+		attachment := attachment{
+			Title:      fmt.Sprintf("❌ Flag \"%s\" deleted", key),
+			Color:      colorDeleted,
+			FooterIcon: goFFLogo,
+			Footer:     microsoftteamsFooter,
+		}
+		attachments = append(attachments, attachment)
+	}
+	return attachments
+}
+
+func convertUpdatedFlagsToMicrosoftTeamsMessage(diffCache notifier.DiffCache) []attachment {
+	attachments := make([]attachment, 0)
+	for key, value := range diffCache.Updated {
+		attachment := attachment{
+			Title:      fmt.Sprintf("✏️ Flag \"%s\" updated", key),
+			Color:      colorUpdated,
+			FooterIcon: goFFLogo,
+			Footer:     microsoftteamsFooter,
+			Fields:     []Field{},
+		}
+
+		changelog, _ := diff.Diff(value.Before, value.After, diff.AllowTypeMismatch(true))
+		for _, change := range changelog {
+			if change.Type == "update" {
+				value := fmt.Sprintf("%s => %s", render.Render(change.From), render.Render(change.To))
+				short := len(value) < longMicrosoftTeamsAttachment
+				attachment.Fields = append(
+					attachment.Fields,
+					Field{Title: strings.Join(change.Path, "."), Short: short, Value: value},
+				)
+			}
+		}
+
+		sort.Sort(ByTitle(attachment.Fields))
+
+		attachments = append(attachments, attachment)
+	}
+	return attachments
+}
+
+func convertAddedFlagsToMicrosoftTeamsMessage(diff notifier.DiffCache) []attachment {
+	attachments := make([]attachment, 0)
+	for key := range diff.Added {
+		attachment := attachment{
+			Title:      fmt.Sprintf("🆕 Flag \"%s\" created", key),
+			Color:      colorAdded,
+			FooterIcon: goFFLogo,
+			Footer:     microsoftteamsFooter,
+		}
+		attachments = append(attachments, attachment)
+	}
+	return attachments
+}
+
+func attachmentsToSections(attachments []attachment) []Section {
+	sections := make([]Section, len(attachments))
+	for i, att := range attachments {
+		facts := make([]Fact, len(att.Fields))
+		for j, field := range att.Fields {
+			facts[j] = Fact(field)
+		}
+
+		sections[i] = Section{
+			Title: att.Title,
+			Color: att.Color,
+			Facts: facts,
+		}
+	}
+	return sections
+}
+
+type AdaptiveCard struct {
+	Type     string    `json:"@type"`
+	Context  string    `json:"@context"`
+	Summary  string    `json:"summary"`
+	Sections []Section `json:"sections"`
+}
+
+type Section struct {
+	Title string `json:"title,omitempty"`
+	Text  string `json:"text,omitempty"`
+	Color string `json:"color,omitempty"`
+	Facts []Fact `json:"facts,omitempty"`
+}
+
+type Fact struct {
+	Title string `json:"title"`
+	Value string `json:"value"`
+	Short bool   `json:"short"`
+}
+type attachment struct {
+	Title      string  `json:"title"`
+	Color      string  `json:"color"`
+	FooterIcon string  `json:"footer_icon"`
+	Footer     string  `json:"footer"`
+	Fields     []Field `json:"fields,omitempty"`
+}
+
+type Field struct {
+	Title string `json:"title"`
+	Value string `json:"value"`
+	Short bool   `json:"short"`
+}
+
+type ByTitle []Field
```

Comment:
```suggestion
type ByTitle []Field
```

**🎯 suggestion:** ‏`ByTitle` does not have to be public I would suggest to put this struct as private.